src/Aspire.Dashboard/Utils/DashboardUrls.cs

```diff
@@ -143,10 +145,12 @@ public static string LoginUrl(string? returnUrl = null, string? token = null)
 
     public static string SetLanguageUrl(string language, string redirectUrl)
     {
-        var url = "/api/set-language";
+        var url = $"{BasePath}api/set-language";
         url = QueryHelpers.AddQueryString(url, "language", language);
         url = QueryHelpers.AddQueryString(url, "redirectUrl", redirectUrl);
 
         return url;
     }
+
+    internal static void SetBasePath(string pathBase) => BasePath = pathBase;
```

Comment:
This doesn't seem like an optimal way to update `BasePath` to me. If `DashboardUrls` is going to store this state, it should be turned into a service and injected.

Reply:
I am following the `SetLanguageUrl` method above

Reply:
It’s different, as SetLanguageUrl does not set state. It only returns the url for the set language url path.

Reply:
OK. Is this something that will block the feature from being accepted?

Reply:
See the comment below, DashboardUrls shouldn't be changed at all